cmd/templ/generatecmd/eventhandler.go

```diff
@@ -257,6 +263,47 @@ func (h *FSEventHandler) generate(ctx context.Context, fileName string) (goUpdat
 	return goUpdated, textUpdated, parsedDiagnostics, err
 }
 
+func mapFormatterError(err error, sourceMap *parser.SourceMap) (error, bool) {
+	line, col, suffix, err2 := parseFormatterError(err.Error())
+	if err2 != nil {
+		return nil, false
+	}
+
+	// The sourceMap positions are off by one because of the package.
+	pos, ok := sourceMap.SourcePositionFromTarget(line-1, col)
+	if !ok {
+		return nil, false
+	}
+	return fmt.Errorf("%d:%d%s", pos.Line+1, pos.Col, suffix), true
+}
+
+func parseFormatterError(str string) (errLine, errCol uint32, suffix string, err error) {
```

Comment:
I think it's better to inspect the error type.

https://go.dev/play/p/YR6TgOR9XeR

Reply:
That is way better, thank you. I updated the code.